cranelift-codegen/meta-python/isa/x86/encodings.py

```diff
@@ -342,9 +342,15 @@ def enc_i32_i64_ld_st(inst, w_bit, recipe, *args, **kwargs):
 X86_32.enc(base.copy_special, *r.copysp(0x89))
 
 # Stack-slot-to-the-same-stack-slot copy, which is guaranteed to turn
-# into a no-op.
+# into a no-op.  Ideally we could to make this encoding available for
+# all types, and write `base.copy_nop.any`, but it appears that the
+# controlling type variable must not polymorphic.  So we make do with
+# the following limited set, and guard the generating transformation in
+# regalloc/reload.rs accordingly.
 X86_64.enc(base.copy_nop.i64, r.stacknull, 0)
 X86_64.enc(base.copy_nop.i32, r.stacknull, 0)
+X86_64.enc(base.copy_nop.i16, r.stacknull, 0)
+X86_64.enc(base.copy_nop.i8,  r.stacknull, 0)
```

Comment:
What about boolean types? Do we need to add an encoding for b1 and b8?

Reply:
From the point of view of "will CL assert/fail", no, we don't need them, because the transformation done by reload.rs is now limited to types satisfying `is_int` or `is_float`, and all targets (x86_64, x86_32, rv32, rv64) now have encodings for those types.

From the point of view of "are we missing optimisations", maybe.  But it seems unlikely to me, assuming that most boolean values are consumed by an "if" as soon as they are created.  On that basis, I omitted them.  If profiling later shows evidence that we need these cases, then we can add them.  I haven't seen any need for them in the profiling I've done so far.